Framework/API/test/NearestNeighboursTest.h

```diff
@@ -225,52 +204,25 @@ class NearestNeighboursTestPerformance : public CxxTest::TestSuite {
 
 public:
   void testUsingRadius() {
-    Instrument_sptr instrument = boost::dynamic_pointer_cast<Instrument>(
+    const auto ws = makeWorkspace(1, 18);
+    ws->setInstrument(
         ComponentCreationHelper::createTestInstrumentCylindrical(2));
-    const ISpectrumDetectorMapping spectramap =
-        NearestNeighboursTest::buildSpectrumDetectorMapping(1, 18);
-    // Default parameter map.
-    ParameterMap_sptr pmap(new ParameterMap());
-    // Parameterized instrument
-    Instrument_sptr m_instrument(new Instrument(instrument, pmap));
 
     // Create the NearestNeighbours object directly.
-    NearestNeighbours nn(m_instrument, spectramap);
+    NearestNeighbours nn(8, ws->spectrumInfo(), getSpectrumNumbers(*ws));
     for (size_t i = 0; i < 2000; i++) {
       nn.neighboursInRadius(1, 5.0);
     }
   }
 
-  void testUsingDefault() {
-    Instrument_sptr instrument = boost::dynamic_pointer_cast<Instrument>(
-        ComponentCreationHelper::createTestInstrumentCylindrical(2));
-    const ISpectrumDetectorMapping spectramap =
-        NearestNeighboursTest::buildSpectrumDetectorMapping(1, 18);
-    // Default parameter map.
-    ParameterMap_sptr pmap(new ParameterMap());
-    // Parameterized instrument
-    Instrument_sptr m_instrument(new Instrument(instrument, pmap));
-
-    // Create the NearestNeighbours object directly.
-    NearestNeighbours nn(m_instrument, spectramap);
-    for (size_t i = 0; i < 2000; i++) {
-      nn.neighboursInRadius(1, 0.0);
-    }
-  }
-
   void testUsingNumberOfNeighbours() {
-    Instrument_sptr instrument = boost::dynamic_pointer_cast<Instrument>(
+    const auto ws = makeWorkspace(1, 18);
+    ws->setInstrument(
         ComponentCreationHelper::createTestInstrumentCylindrical(2));
-    const ISpectrumDetectorMapping spectramap =
-        NearestNeighboursTest::buildSpectrumDetectorMapping(1, 18);
-    // Default parameter map.
-    ParameterMap_sptr pmap(new ParameterMap());
-    // Parameterized instrument
-    Instrument_sptr m_instrument(new Instrument(instrument, pmap));
 
     // Create the NearestNeighbours object directly.
     for (size_t i = 0; i < 2000; i++) {
-      NearestNeighbours nn(8, m_instrument, spectramap);
+      NearestNeighbours nn(8, ws->spectrumInfo(), getSpectrumNumbers(*ws));
       nn.neighbours(1);
     }
   }
```

Comment:
No, but this test seems a bit useless anyway. No real code would create the `NearestNeighbours` object and use it to get neighbours only for a single detector.